src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1508,6 +1534,22 @@ string targetDatabase
             return result;
         }
 
+        /// <summary>
+        /// Helper method to create result object from exception
+        /// </summary>
+        /// <param name="ex"></param>
+        private string GetErrorMessageFromException(Exception e)
```

Comment:
Would suggest putting this in a more common location (and maybe combining with https://github.com/microsoft/sqltoolsservice/blob/38f33fa211086d0def50b016dd6940b942ac9255/src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs#L167)

Reply:
Made an extension